frontend/src/components/FilterDrawer.vue

```diff
@@ -258,9 +269,45 @@ const updateMap = async () => {
   selectedTreeItems.value.forEach((item) => {
     selectedProcesses.value.push(item)
   })
-  await nextTick()
+  // await nextTick()
   filter()
 }
+const debouncedFilter = () => {
+  clearTimeout(debounceTimeout.value)
+  debounceTimeout.value = setTimeout(() => {
+    filter()
+  }, 300) // 300ms debounce
```

Comment:
debouncing the search is a good idea. I wonder if it would have been easier to use a different event on the search instead, but this looks fine too!